datacapture/src/main/java/com/google/android/fhir/datacapture/views/OptionSelectDialogFragment.kt

```diff
@@ -0,0 +1,360 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.android.fhir.datacapture.views
+
+import android.app.Dialog
+import android.os.Bundle
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.Button
+import android.widget.CheckBox
+import android.widget.EditText
+import android.widget.RadioButton
+import androidx.appcompat.app.AlertDialog
+import androidx.core.widget.doAfterTextChanged
+import androidx.fragment.app.DialogFragment
+import androidx.fragment.app.activityViewModels
+import androidx.lifecycle.lifecycleScope
+import androidx.recyclerview.widget.DiffUtil
+import androidx.recyclerview.widget.LinearLayoutManager
+import androidx.recyclerview.widget.ListAdapter
+import androidx.recyclerview.widget.RecyclerView
+import com.google.android.fhir.datacapture.R
+import java.util.concurrent.atomic.AtomicInteger
+import kotlinx.coroutines.flow.collect
+import kotlinx.coroutines.launch
+
+internal class OptionSelectDialogFragment(
+  val title: String,
+  val config: Config,
+) : DialogFragment() {
+
+  /** Configures this [OptionSelectDialogFragment]. */
+  data class Config(
+    /** Whether multi-select will be enabled or not. */
+    val multiSelect: Boolean,
+    /** Whether the "other" option will be exposed for free-form text input. */
+    val addOtherOptions: Boolean,
```

Comment:
```suggestion
    val canAddOtherOptions: Boolean,
```

or maybe `otherOptionsAllowed`?